packages/botonic-core/src/models/legacy-types.ts

```diff
@@ -232,8 +232,8 @@ export interface PluginPostRequest extends BotResponse {
 }
 
 export interface Plugin {
-  post(request: PluginPostRequest): void
-  pre(request: PluginPreRequest): void
+  post?(request: PluginPostRequest): void | Promise<void>
+  pre?(request: PluginPreRequest): void | Promise<void>
```

Comment:
The way it's done right now, the plugins require to have both `pre` and `post` methods because they're always called within the `runPlugins` method, so the `?` should not be added here unless the behaviour is changed in the `runPlugins` method to call the `pre` and `post` methods only if they exist.

Reply:
fixed @asastre 👌